### ISSUE TITLE
MSBuildWorkspace fails to load metadatareferences for a project with a non-existent default configuration

### ISSUE THREAD

**Issue by mhgamework** [dotnet/roslyn#5389]
I loaded a csproj using MSBuildWorkspace but the project.MetadataReferences was empty. The project compiles and runs fine in VS2015 and has numerous (+15) third party references. I discovered that the project had the default configuration Debug|AnyCPU, which did not exists, so i fixed it to an existing configuration and the project now loads fine.

I do not understand this behaviour, it seems like a bug to me, or that there should be some sort of exception thrown.

For more information, see: http://stackoverflow.com/questions/32634648/compilation-fails-with-roslyn-but-works-with-visual-studio/32650912?noredirect=1#comment53249254_32650912

**Comment by Pilchie** [comment]
Can you construct a repro project with this behavior?

**Comment by mhgamework** [comment]
Yes, I can and have stored the solution in a zip archive. Is there a way I can upload this archive here? At the moment I have shared it on my dropbox. 
https://www.dropbox.com/s/p8j3v3uqh6u1z52/RoslynIssue5389.zip?dl=0

**Comment by Pilchie** [comment]
Thanks!  I've saved a copy of this inside Microsoft if you want to free up the space on your dropbox.

**Comment by leftler** [comment]
I came to post a similar issue. If you would like another repo example See https://github.com/leftler/RoslynIssue I did not realize I could just check `project.MetadataReferences` in my repo I did 

        [TestMethod]
        public async Task TestFailingDummyProgramWithDefaultSettings()
        {
            var ws = Microsoft.CodeAnalysis.MSBuild.MSBuildWorkspace.Create();
            var solution = await ws.OpenSolutionAsync(@"..\FailingProgram\DummyProgram.sln");
            Assert.AreEqual(1, solution.Projects.Count());

            foreach (var project in solution.Projects)
            {
                var comp = await project.GetCompilationAsync();
                Assert.IsTrue(comp.References.Any(x=>x.Display.EndsWith(@"\mscorlib.dll")), "Reference to mscorlib.dll not found found");
            }
        }
